test/distributed/test_c10d_nccl.py

```diff
@@ -231,6 +231,9 @@ def setUp(self):
         # that use TORCH_NCCL_BLOCKING_WAIT will test it as expected.
         os.environ["TORCH_NCCL_ASYNC_ERROR_HANDLING"] = "1"
         # self.num_gpus = torch.cuda.device_count()
+        # To test NONBLOCKING NCCL calls
+        # os.environ["TORCH_NCCL_USE_COMM_NONBLOCKING"] = "1"
```

Comment:
should we make the test run under both modes? otherwise aren't we killing off test coverage for the codepath we actually use in prod?

Reply:
I believe so too, but didn't find a good way yet to do so without copy&paste tests. Any recommendations?

Reply:
In the test script where test_c10d_nccl.py is called, set the env and run twice.

Reply:
OK, added 2 unit tests specifically for non-blocking mode in the stacked PR